miden-lib/asm/sat/layout.masm

```diff
@@ -18,6 +18,9 @@ const.NUM_CREATED_NOTES_PTR=2
 # The memory address at which a pointer to the consumed note being executed is stored.
 const.CURRENT_CONSUMED_NOTE_PTR=3
 
+# The memory address at which a flag indicating whether the transaction is in the prologue is stored.
+const.IN_PROLOGUE_PTR=4
```

Comment:
I think the idea is correct but I wonder if the naming should be different. Specifically, we kind of have 4 different phases of execution:

1. Prologue - in which case `sender` should be all zeros.
2. Note execution - in which case `sender` should be the sender of the currently executing note.
3. Tx script execution (optional) - in which case `sender` should be all zeros.
4. Epilogue - in which case `sender` should be all zeros.

So, I wonder if the flag should be set at the beginning of note execution and then reset before tx script execution. And then we could call it `EXECUTING_NOTES` or something like that.

Alternatively, we could compute the flag on the fly based on the pointer to the currently executing note (i.e., if the pointer is ZERO, we are not executing a note at this time).

Reply:
This makes sense.  The currently implementation is setting the flag during the epilogue which isn't correct.  I think your proposal of using `get_current_consumed_note_ptr` makes sense.  We will need to ensure that we set the pointer to ZERO once all notes have been executed.  As it stands we are not able to implement this as we do not have a handler for steps 2-3, which is where the pointer would need to be set to 0.  I can create an issue to implement it once it becomes possible.

Reply:
Issue created: https://github.com/0xPolygonMiden/miden-base/issues/94